iceoryx_binding_c/source/c_publisher.cpp

```diff
@@ -99,11 +99,31 @@ void iox_pub_deinit(iox_pub_t const self)
     self->~cpp2c_Publisher();
 }
 
-iox_AllocationResult iox_pub_loan_chunk(iox_pub_t const self, void** const chunk, const uint32_t payloadSize)
+bool iox_pub_configure_custom_header(iox_pub_t const self,
+                                     const uint32_t customHeaderSize,
+                                     const uint32_t customHeaderAlignment)
 {
-    auto result = PublisherPortUser(self->m_portData)
-                      .tryAllocateChunk(payloadSize, CHUNK_DEFAULT_PAYLOAD_ALIGNMENT)
-                      .and_then([&chunk](ChunkHeader* h) { *chunk = h->payload(); });
+    auto settingsCheck = iox::mepoo::ChunkSettings::create(0U, 1U, customHeaderSize, customHeaderAlignment);
+    if (settingsCheck.has_error())
+    {
+        return false;
+    }
+
+    self->m_customHeaderSize = customHeaderSize;
+    self->m_customHeaderAlignment = customHeaderAlignment;
+
+    return true;
+}
+
+iox_AllocationResult iox_pub_loan_chunk(iox_pub_t const self,
+                                        void** const chunk,
+                                        const uint32_t payloadSize,
+                                        const uint32_t payloadAlignment)
+{
+    auto result =
+        PublisherPortUser(self->m_portData)
+            .tryAllocateChunk(payloadSize, payloadAlignment, self->m_customHeaderSize, self->m_customHeaderAlignment)
```

Comment:
Works, but probably is suboptimal performance wise to pass it every time to the port (maybe it is optimized, did not check). Ideally the port or the entity creating/allocating the chunk would know the size and alignment. Oh well, it is basically an aligned alloc (at least partly).
It is not likely to change often or at all over the lifetime of the publisher I think.

Reply:
good point, but that's not user facing API and could easily changed afterwards

Reply:
Sure, low prio and user API ans functionality is the main focus. Just an observation, nothing to fix.